src/Layer.js

```diff
@@ -62,35 +48,26 @@ export default class Layer {
         this._checkId(id);
         this._checkSource(source);
         this._checkViz(viz);
-        this._oldDataframes = new Set();
-        this._isInitialized = false;
         this._init(id, source, viz);
     }
 
     _init (id, source, viz) {
         viz._boundLayer = this;
-        this.state = 'init';
-        this._id = id;
-
-        this._emitter = mitt();
-        this._lastViewport = null;
-        this._lastMNS = null;
-        this._integrator = null;
-
         this._context = new Promise((resolve) => {
             this._contextInitialize = resolve;
         });
 
-        this._integratorPromise = new Promise((resolve) => {
-            this._integratorCallback = resolve;
-        });
-
+        this.id = id;
+        this.type = 'custom';
```

Comment:
Why is this added?

Reply:
The idea is to convert `carto.Layer` in a native MGL CustomLayer. I'll add a comment there